### ISSUE TITLE
Commands should be silent by default

### ISSUE THREAD

**Issue by kdave** [btrfs8-revamp/btrfs8-revamp#2]
The common unix utilities do not print any output if the command succeeds. The
behaviour of the btrfs subcommands usually prints some message and the only
way to drop the output is to redirect to /dev/null .

Changing the output has a strong **backward compatibility** impact that needs
to be considered and discussed.

Proposed fix:

* add `--verbose` and `--quiet` command everywhere applicable (create new issue when started)

After some period (1 year) the defaults will change to silent mode.

**Comment by Ferroin** [comment]
I would argue that certain things (mkfs and check immediately come to mind) should always default to providing information.  I do think they should have a `--quiet` option, but I feel that they should be consistent with other equivalent commands, and most mkfs and fsck commands print general diagnostic info unless you tell them not to.

**Comment by moviuro** [comment]
My main inspiration for starting the whole topic was that I use `zfs(8)` on FreeBSD and it feels like a super tool for sysadmins. It behaves as expected for a CLI tool.

However, I'm not familiar with much more. if you tell me that `mkfs` is verbose, then let btrfs' equivalent be verbose too.

The `-v|-q` options look good. However, I think there is much more to verbosity than just verbose/silent.
A scale from `0` to `n` would IMO be better, as follows:
* `0|silent`: complete silence, don't print anything. Exit code is the only info we get
* `1|fatal`: print only fatal errors, exit codes.
* `2|error`: print all errors, exit codes. (e.g. default for `btrfs sub create|delete`)
* `3|info`: print some useful info, all errors, exit codes. (e.g. default for `mkfs`)
* `4|verbose|debug`: lots of stuff, perhaps not needed if we can use `strace` or w/e

This scale should be discussed for all commands.

**Comment by kdave** [comment]
The more finegrained levels of verbosity are a good idea. I'd add some way to print a timestamp before each message (off by default).

**Comment by Ferroin** [comment]
I like the idea of having a scale, although that brings up the question of how to implement it, as there doesn't seem to be much in the way of consistency between projects that do this.  Some use the level of repetition of -v to indicate the level (so -vvv is more verbose than -v), others use a number after an option (for example, -v5 or -l2 (-l seems to be popular for stuff that's more likely to be redirected to a log file)), others have dedicated options for each level.

**Comment by moviuro** [comment]
The current state on `btrfs(8)` AFAICT is the use of `-v`.
From my personal experience, I highly prefer `-vvv` over `-v3`.

It may also be a good idea to support both a short and a long option:
* `--verbosity=silent|fatal|error|info|debug`
* `--verbosity={0..4}`
* `--silent|--fatal|--error|--info|--debug`

**Comment by Ferroin** [comment]
I agree about having long options, they make scripts clearer and easier to understand, although I prefer the first and second variants you listed to having separate options for every level.

**Comment by Ferroin** [comment]
I just realized, `btrfs subvolume list` already has a `-q` option, although the choice of `-q` there seems kind of arbitrary and somewhat odd (It's for listing parent UUID's of subvolumes, so `-p` might make more sense, and I'm somewhat confused about why that wasn't chosen given that `-p` isn't an option for `subvolume list`).
